activerecord/lib/active_record/autosave_association.rb

```diff
@@ -176,7 +176,8 @@ def define_non_cyclic_method(name, &block)
         # check if the save or validation methods have already been defined
         # before actually defining them.
         def add_autosave_association_callbacks(reflection)
-          save_method = :"autosave_associated_records_for_#{reflection.name}"
+          method_suffix = reflection.options[:class_name].try(:underscore)
```

Comment:
@arthurnn sorry but I didn't understand what do you mean by `join`. If you are talking about the `Array#join` I would say this is not an array, this is just a `String` or `Nil`. Do you want me to do `string.split('::').join('_')`. Can you please give an example?